application/F3DStarter.cxx

```diff
@@ -273,29 +281,107 @@ int F3DStarter::Start(int argc, char** argv)
 //----------------------------------------------------------------------------
 void F3DStarter::LoadFile(f3d::loader::LoadFileEnum load)
 {
+  if (this->Internals->LoadedFile)
+  {
+    // When loading a file after another, store the changed options
+    // into the dynamic options and use these dynamic option as the default
+    // for loading the next file while still applying file specific options on top of it
+
+    // Recover previous options from the engine
+    f3d::options previousOptions = this->Internals->Engine->getOptions();
+
+    // Detect changed options and apply the change to the dynamic options
+    // options names and types are shared between options instance
+    std::vector<std::string> optionNames = this->Internals->DynamicOptions.getNames();
+    for (auto name : optionNames)
+    {
+      switch (this->Internals->DynamicOptions.getType(name))
+      {
+        case f3d::options::Type::BOOL:
+        {
+          bool val;
+          previousOptions.get(name, val);
+          if (this->Internals->FileOptions.getAsBool(name) != val)
```

Comment:
why not creating a `f3d::options::copyFrom(const f3d::options& opt, const std::string& name)` to copy the variant instead of this switch case block?

Reply:
If F3D needs it, libf3d users can need it, so yes I think it would be useful.

Reply:
agreed.

Reply:
actually, copyFrom is not what we want as we only want to copy if a condifition is true.

It boils down to:

if the option value in the engine is different from the option passed to the engine, copy this value to the dynamic options.

There is three options involved here, and this is what makes it work.

of course I could add a `f3d::options::copyToIfDifferentWIth(f3d::options& to, const f3d::options& with, const std::string& name)` but that would actually be overspecific.

maybe I can just expose the std::variant in the api ?